### ISSUE TITLE
Static Web App deploy fails with no error details when using C# Azure Function as API with "FunctionsStartup" configuration

### ISSUE THREAD

**Issue by nazar-kuzo** [Azure/static-web-apps#244]
![image](https://user-images.githubusercontent.com/13677730/102932483-0b9e0800-44a9-11eb-9fa7-942ccfbda104.png)

Having deploy of the same version without Startup file (for dependency injection) it deploys with no error

**Comment by plamber** [comment]
Hello,
we are experiencing the same issue. Support for dependency injection is key for some projects.

br,

**Comment by anthonychu** [comment]
@plamber Is it possible to share your code or a sample that reproduces the problem?

/cc @annaji-msft

**Comment by plamber** [comment]
Hi @anton-kokorev,
you can reproduce it by following the standard provisioning process.

Afterwards, you just need to rename the Github workflow yml file name to something else. Starting from that point, the link is no more up-to-date in Azure.

You can still execute the workflows though.

br,
Patrick

**Comment by nazar-kuzo** [comment]
@plamber, were you intentionally mentioning @anton-kokorev, or it was a question to me?

**Comment by plamber** [comment]
Hi @nazar-kuzo,
@anthonychu  asked me how to reproduce it. I tried to answer his question.

br,
Patrick

**Comment by plamber** [comment]
Hi @nazar-kuzo,
sorry. Just recognized that I answered the wrong thread. 

@anton-kokorev: unfortunately, I can't share my repository but you should be able to reproduce the error with [this](https://github.com/AzureAD/microsoft-identity-web/tree/master/ProjectTemplates/templates/Functions-CSharp) project template. We inject the Microsoft Identity Web library and some other internal libraries in the Startup.

br,
Patrick

**Comment by davidobrien1985** [comment]
@anthonychu @annaji-msft this is still an issue. I just stumbled upon the same error when adding dependency injection to our static web app. This is following the exact same process like in any other Azure Function.
On SWA this fails though. Major blocker now.

Here's our `Startup.cs`:

```
using System;
using app;
using Microsoft.AspNetCore.Authentication.JwtBearer;
using Microsoft.Azure.Functions.Extensions.DependencyInjection;
using Microsoft.Extensions.DependencyInjection;

[assembly: FunctionsStartup(typeof(Startup))]
namespace app
{
    public class Startup : FunctionsStartup
    {
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddAuthentication(options =>
            {
                options.DefaultAuthenticateScheme =  JwtBearerDefaults.AuthenticationScheme;
                options.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
            }).AddJwtBearer(options =>
            {
                options.Authority = Environment.GetEnvironmentVariable("auth0Authority");
                options.Audience = System.Environment.GetEnvironmentVariable("auth0Audience");
            });
        }

        public override void Configure(IFunctionsHostBuilder builder)
        {
            ConfigureServices(builder.Services);
        }
    }
}
```

**Comment by nazar-kuzo** [comment]
@davidobrien1985 run away from this approach to more complex but predictable separate deployment of UI and API separately, since 2 years of no resolution for this bug should tell you something 😉

**Comment by davidobrien1985** [comment]
Yeah, decided to run the Functions ourselves now, and will likely also move the frontend to a different service, probably just a Storage Account. Pity.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
